staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,258 @@
+func (reqMgmt *requestManagementSystem) OnDelete(obj interface{}) {
+	reqMgmt.triggerReload()
+}
+
+func (reqMgmt *requestManagementSystem) Run(stopCh <-chan struct{}) error {
+	defer reqMgmt.configQueue.ShutDown()
+	klog.Info("Starting reqmgmt config controller")
+	if ok := cache.WaitForCacheSync(stopCh, reqMgmt.plInformerSynced, reqMgmt.fsInformerSynced); !ok {
+		return fmt.Errorf("Never achieved initial sync")
+	}
+	klog.Info("Running reqmgmt config worker")
+	wait.Until(reqMgmt.runWorker, time.Second, stopCh)
+	klog.Info("Shutting down reqmgmt config worker")
+	return nil
+}
+
+func (reqMgmt *requestManagementSystem) runWorker() {
+	for reqMgmt.processNextWorkItem() {
+	}
+}
+
+func (reqMgmt *requestManagementSystem) processNextWorkItem() bool {
+	obj, shutdown := reqMgmt.configQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	func(obj interface{}) {
+		defer reqMgmt.configQueue.Done(obj)
+		if !reqMgmt.syncOne() {
+			reqMgmt.configQueue.AddRateLimited(obj)
+		} else {
+			reqMgmt.configQueue.Forget(obj)
+		}
+	}(obj)
+
+	return true
+}
+
+// syncOne attempts to sync all the config for the reqmgmt filter.  It
+// either succeeds and returns `true` or logs an error and returns
+// `false`.
+func (reqMgmt *requestManagementSystem) syncOne() bool {
+	all := labels.Everything()
+	newPLs, err := reqMgmt.plLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list PriorityLevelConfiguration objects: %s", err.Error())
+		return false
+	}
+	newFSs, err := reqMgmt.fsLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list FlowSchema objects: %s", err.Error())
+		return false
+	}
+	reqMgmt.digestConfigObjects(newPLs, newFSs)
+	return true
+}
+
+func (reqMgmt *requestManagementSystem) digestConfigObjects(newPLs []*rmtypesv1a1.PriorityLevelConfiguration, newFSs []*rmtypesv1a1.FlowSchema) {
+	oldRMState := reqMgmt.curState.Load().(*requestManagementState)
+	var shareSum float64
+	newRMState := &requestManagementState{
+		priorityLevelStates: make(map[string]*priorityLevelState),
+	}
+	newlyQuiescent := make([]*priorityLevelState, 0)
+	for _, pl := range newPLs {
+		state := oldRMState.priorityLevelStates[pl.Name]
+		if state == nil {
+			state = &priorityLevelState{
+				config: pl.Spec,
+			}
+		} else {
+			if state.config.Exempt {
+				newRMState.priorityLevelStates[pl.Name] = state
+				continue
+			}
+
+			oState := *state
+			state = &oState
+			state.config = pl.Spec
+			if state.emptyHandler != nil { // it was undesired, but no longer
+				klog.V(3).Infof("Priority level %s was undesired and has become desired again", pl.Name)
+				state.emptyHandler = nil
+				state.queues.Quiesce(nil)
+			}
+		}
+		shareSum += float64(state.config.AssuredConcurrencyShares)
+		newRMState.priorityLevelStates[pl.Name] = state
+	}
+
+	fsSeq := make(rmtypesv1a1.FlowSchemaSequence, 0, len(newFSs))
+	for _, fs := range newFSs {
+		reqMgmt.syncFlowSchemaStatus(fs, newRMState.priorityLevelStates[fs.Spec.PriorityLevelConfiguration.Name] == nil)
```

Comment:
There could be a lot of apiservers doing this.  I do not favor many processes all trying to report this bit of information.  A better solution would be to create a controller (that lives in controllers-manager) for that purpose.  That can wait for a later PR.